### PR TITLE
add extrinsic index to payload in gen_dna helper function

### PR DESCRIPTION
**Pull request by cc-michael-x**

### CODE REVIEW

**Review by sacha-l — APPROVED** [review]

**Comment by sacha-l** [comment]
Thanks for your contribution @cc-michael-x 🙏🏻  👍🏻

File: pallets/kitties/src/lib.rs [hunk 1]
```diff
@@ -341,6 +342,9 @@ pub mod pallet {
 			let new_cnt = Self::kitty_cnt().checked_add(1)
 				.ok_or(<Error<T>>::KittyCntOverflow)?;
 
+			// Check if the kitty does not already exist in our storage map
+			ensure!(Self::kitties(&kitty_id) == None, <Error<T>>::KittyExists);
```
Comment (sacha-l) [inline pallets/kitties/src/lib.rs:346]:
```suggestion
			ensure!(Self::kitties(&kitty_id) == None, <Error<T>>::KittyNotExist);
```
Will put this in a separate commit. Didn't catch this in time.

Comment (cc-michael-x) [reply pallets/kitties/src/lib.rs:346]: This should be `KittyExists`, not `KittyNotExist`. Reason being, the user is attempting to create a kitty that already exists, which is thrown as an error `KittyExists`.

**Comment by cc-michael-x** [comment]
Please review and merge this pull request to update the tutorial documentation related to this PR. [substrate-docs](https://github.com/substrate-developer-hub/substrate-docs/pull/777)
